### DISCUSSION TITLE
https://blog.tidy-intelligence.com/posts/data-storage-comparison/

### DISCUSSION THREAD

**Discussion by giscus** [tidy-intelligence/blog#2] [Announcements]
# https://blog.tidy-intelligence.com/posts/data-storage-comparison/

A comparison of popular open-source data storage technologies using R and Python.

https://blog.tidy-intelligence.com/posts/data-storage-comparison/

**Comment by iangow** [comment]
I suspect that DuckDB is more aware of pandas than vice versa. So you might want to use DuckDB's functions to interact with pandas. For one, "round-tripping" the data results in no loss in terms of types.

``` python
duckdb.sql('CREATE TABLE data AS SELECT * FROM data_python')
duckdb.sql('SELECT * FROM data')
data_python_duckdb = duckdb.sql('SELECT * FROM data').df()
data_python_duckdb.info()
```

**Reply by christophscheuch** [reply]
Thank you for the suggestion, I'll include that in the post :pray:
